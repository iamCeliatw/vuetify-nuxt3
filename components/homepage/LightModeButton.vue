<template lang="pug">
.container
  .btn
    input(type="checkbox" name="check" id="check" @change="changeMode")
    label(for="check")
      .box
        .ball(:style="{transform: colorMode.value === 'dark' ? 'translatex(100%)' : 'translatex(0%)'}")
        .scenary
          HomepageSvgMoonIcon
          HomepageSvgSunIcon
</template>

<script lang="ts" setup>
const colorMode = useColorMode();
const changeMode = () => {
  console.log(colorMode);
  colorMode.preference = colorMode.value === "dark" ? "light" : "dark";
};
</script>

<style lang="sass" scoped>
@import "@/assets/mixin.sass"
$scale:55px
$time: 1s
.container
  display: flex
  flex-direction: column
  width: 80%
  @media (min-width:500px)
    width: 50%

input[type="checkbox"]
  display: none


.box
  @include common(flex,column,center,null,$scale,calc($scale/2),var(--box-color))
  position: relative
  border-radius: calc($scale/4)
  cursor: pointer


.ball
  @include common(null,null,null,null,calc($scale/2),calc($scale/2),grey)
  position: absolute
  border-radius: 50%
  border: 3px solid black


.scenary
  @include common(flex, row, space-between,null,null,null,null)
  padding: 2px 10px 0px 10px
  svg
    width: calc($scale / 3)
</style>
