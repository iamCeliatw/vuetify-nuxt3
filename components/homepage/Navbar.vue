<template lang="pug">
nav.navbar__wrapper 
  .navbar__container
    .navbar__title 
      //- p Celia's note
    .button__section 
      ul.navigate__section
        li(:class="{ 'active-route': router.currentRoute.value.path === nav.url }" v-for="nav in navigate" :key="nav", @click="() => router.push(nav.url)" ) {{ nav.name }}
      //- .github__icon 
      //-   img(src="/github.png")
      .light__button
        HomepageLightModeButton
</template>

<script lang="ts" setup>
const colorMode = useColorMode();
const router = useRouter();
const navigate = ref([
  { name: "post", url: "/" },
  { name: "about", url: "/about" },
  { name: "category", url: "/category" },
]);
</script>

<style lang="sass" scoped>
@import "../../assets/breakpoint"

.navbar__wrapper
  width: auto
  max-width: 800px
  height: 90px
  display: flex
  justify-content: center
  margin: 0 auto
  @include breakPoint(size-768 ,null)
    width: 90%
  @include breakPoint(size-512 ,null)
    width: 100%
.navbar__container
  padding: 0 20px
  display: flex
  width: 100%
  height: 100%
  justify-content: space-between
  align-items: center
  @include breakPoint(size-768 ,null)
    // padding: 0 10p
.navigate__section
  display: flex
  list-style-type: none
  li
    margin-right: 20px
    font-weight: 500
    font-size: 20px
    cursor: pointer
    height: 40px
    line-height: 40px
    transition: all .2s
    &:hover
      border-bottom: 2px solid
    @include breakPoint(size-768 ,null)
      font-size: 16px
.button__section
  display: flex
  align-items: center
  gap: 10px
  @include breakPoint(size-512 ,null)
    width: 100%
    justify-content: space-between
.github__icon
  width: 30px
  height: 30px
  border-radius: 50%
  img
    width: 100%
    height: 100%
    object-fit: cover
.active-route
  border-bottom: 2px solid
.navbar__title
  p
    font-size: 26px
    @include breakPoint(size-768 ,null)
      font-size: 20px
    @include breakPoint(size-512 ,null)
      display: none

.light__button
  width: auto
</style>
