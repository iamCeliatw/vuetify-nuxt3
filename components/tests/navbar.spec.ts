import { mount, shallowMount } from "@vue/test-utils";
import Navbar from "../homepage/Navbar.vue";
import LightModeButton from "../homepage/LightModeButton.vue";
vi.stubGlobal("useColorMode", vi.fn());
vi.stubGlobal(
  "useRouter",
  vi.fn(() => ({
    currentRoute: {
      value: {
        path: "/",
      },
    },
  }))
);

describe("Navbar", () => {
  it("should render correctly", () => {
    const wrapper = shallowMount(Navbar, {
      stubs: {
        //   HomepageNavbar: true,
        LightModeButton: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
