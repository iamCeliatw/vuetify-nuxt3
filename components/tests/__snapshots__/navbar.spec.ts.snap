// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navbar > should render correctly 1`] = `
"<nav data-v-0e20c101="" class="navbar__wrapper">
  <div data-v-0e20c101="" class="navbar__container">
    <div data-v-0e20c101="" class="navbar__title"></div>
    <div data-v-0e20c101="" class="button__section">
      <ul data-v-0e20c101="" class="navigate__section">
        <li data-v-0e20c101="" class="active-route">post</li>
        <li data-v-0e20c101="" class="">about</li>
        <li data-v-0e20c101="" class="">category</li>
      </ul>
      <div data-v-0e20c101="" class="light__button">
        <light-mode-button-stub data-v-0e20c101=""></light-mode-button-stub>
      </div>
    </div>
  </div>
</nav>"
`;
